modules/openapi-generator/src/main/resources/scala-sttp-client/api.mustache

```diff
@@ -0,0 +1,65 @@
+{{>licenseInfo}}
+package {{package}}
+
+{{#imports}}
+import {{import}}
+{{/imports}}
+import {{mainPackage}}.core._
+import {{mainPackage}}.core.CollectionFormats._
+import sttp.client._
+import sttp.model.Method
+import sttp.client.json4s.SttpJson4sApi
+
+{{#operations}}
+object {{classname}} {
+
+  def apply(baseUrl: String = "{{{basePath}}}")(implicit serializer: SttpSerializer) = new {{classname}}(baseUrl)
+}
+
+class {{classname}}(baseUrl: String)(implicit serializer: SttpSerializer) {
+
+  import serializer._
+
+  /*
+   * Helper to handle Optional header parameters
+   **/
+  implicit class optionalParams(request: RequestT[Identity, Either[String, String], Nothing]) {
+   def header( header: String, optValue: Option[Any]): RequestT[Identity, Either[String, String], Nothing] = {
+      optValue.map( value => request.header(header, value.toString)).getOrElse(request)
+    }
+  }
+
+{{#operation}}
+{{#javadocRenderer}}
+{{>javadoc}}
+{{/javadocRenderer}}
+  def {{operationId}}({{>methodParameters}}): RequestT[Identity, Either[ResponseError[Exception], {{>operationReturnType}}], Nothing] =
+    basicRequest
+      .method(Method.{{httpMethod.toUpperCase}}, uri"$baseUrl{{{path}}}{{#queryParams.0}}?{{#queryParams}}{{baseName}}=${{{paramName}}}{{^-last}}&{{/-last}}{{/queryParams}}{{/queryParams.0}}")
+      .contentType({{#consumes.0}}"{{{mediaType}}}"{{/consumes.0}}{{^consumes}}"application/json"{{/consumes}}){{#headerParams}}
+      .header({{>paramCreation}}){{/headerParams}}{{#formParams.0}}
+      .body(Map({{#formParams}}
+        {{>paramFormCreation}},{{/formParams}}
+      )){{/formParams.0}}{{#bodyParam}}
+      .body({{paramName}}){{/bodyParam}}
+      .response(asJson[{{>operationReturnType}}])
+
+
+{{#x-skip-this}}
+
+
+      {{#authMethods}}{{#isApiKey}}.withApiKey(apiKey, "{{keyParamName}}", {{#isKeyInQuery}}QUERY{{/isKeyInQuery}}{{#isKeyInHeader}}HEADER{{/isKeyInHeader}}{{#isKeyInCookie}}COOKIE{{/isKeyInCookie}})
+      {{/isApiKey}}{{#isBasic}}{{#isBasicBasic}}.withCredentials(basicAuth){{/isBasicBasic}}{{#isBasicBearer}}.withCredentials(bearerToken){{/isBasicBearer}}{{/isBasic}}{{/authMethods}}
+
+{{#responses}}{{^isWildcard}}{{#dataType}}.with{{>responseState}}Response[{{dataType}}]({{code}})
+      {{/dataType}}{{^dataType}}.with{{>responseState}}Response[Unit]({{code}})
+      {{/dataType}}{{/isWildcard}}{{/responses}}{{#responses}}{{#isWildcard}}{{#dataType}}.withDefault{{>responseState}}Response[{{dataType}}]
+      {{/dataType}}{{^dataType}}.withDefault{{>responseState}}Response[Unit]
+      {{/dataType}}{{/isWildcard}}{{/responses}}
+{{/x-skip-this}}
+
+{{/operation}}
+
+}
+
```

Comment:
for common types like `RequestT[Identity, Either[ResponseError[Exception], Unit], Nothing]` it will be convenient to have type alias

Reply:
added